### ISSUE TITLE
Product image incorrect when accessing product variation through URL

### ISSUE THREAD

**Issue by rudolfl** [woothemes/woocommerce#50081]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

Assume product with variations where each variation has its own mage (color variation in my case). 
Assume one accesses product page and target certain variation the the URL. Here is an example on my site
https://www.filliescollection.com.au/large-silk-flower-headband-fascinator/?attribute_pa_colors=orange

Where would this scenario occur? Those URLs are submitted per variation in Google product feed (or any other product feed for that matter). Customers then see correct product in ads and search results and expect to see same image when they click on the ad.

I am using Avada theme, but exactly same behaviour is observed with stock StoreFront theme.

### Expected behavior

Main image is changed to the variation image. 

### Actual behavior

Main image stay unchanged. First thumbnail is changed to the correct image, however clicking on it does not change main image. Clicking on other thumbnails show them correctly in main image, but clicking first thumbnail brings up only product main image. In order to get it to behave normally, one has to select another variation first.

### Steps to reproduce

https://www.filliescollection.com.au/large-silk-flower-headband-fascinator/?attribute_pa_colors=orange
Observe wrong variation image being displayed. Click on first thumbnail (orange), observe no change in main image.

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 9.1.4
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.8.1
Log Directory Writable: ✔
WP Version: 6.6.1
WP Multisite: –
WP Memory Limit: 512 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: ✔

### Server Environment ###

Server Info: LiteSpeed
PHP Version: 8.3.9
PHP Post Max Size: 32 MB
PHP Time Limit: 300
PHP Max Input Vars: 1680
cURL Version: 7.61.1
OpenSSL/1.1.1k

SUHOSIN Installed: –
MySQL Version: 10.6.18-MariaDB
Max Upload Size: 32 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

acf-field: 7
acf-field-group: 4
attachment: 2946
awb_off_canvas: 1
fusion_element: 26
fusion_form: 4
fusion_icons: 5
fusion_tb_layout: 12
fusion_tb_section: 34
nav_menu_item: 31
page: 12
post: 2
product: 49
product_variation: 435
revision: 681
shop_order: 21
shop_order_refund: 1
themefusion_elastic: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (19) ###

Advanced Custom Fields PRO: by WP Engine – 6.3.4
bunny.net: by bunny.net – 2.2.14
Code Profiler Pro: by Jerome Bruandet ~ NinTechNet Ltd. – 1.6.10
Disable Comments RB: by rbPlugins – 1.0.23
Instant Indexing: by Rank Math – 1.1.18
Avada Builder: by ThemeFusion – 3.11.9
Avada Core: by ThemeFusion – 5.11.9
LiteSpeed Cache: by LiteSpeed Technologies – 6.2.0.1
Image optimization service by Optimole: by Optimole – 3.13.3
Payment Plugins for PayPal WooCommerce: by Payment Plugins
support@paymentplugins.com – 1.0.50

RL Custom Order Print: by Rudolf Ladyzhenskii – 1.0
Rank Math SEO PRO: by Rank Math – 3.0.67
Rank Math SEO: by Rank Math – 1.0.224
Product Feed PRO for WooCommerce: by AdTribes.io – 13.3.4
WooCommerce Advanced Bulk Edit: by George Iron & Yas G. – 5.4.2
WooCommerce Eway Payment Gateway: by Eway – 3.7.2
PDF Invoices & Packing Slips for WooCommerce: by WP Overnight – 3.8.6
WooCommerce: by Automattic – 9.1.4
WP Meteor: by Aleksandr Guidrevitch – 3.4.4

### Inactive Plugins (5) ###

Conversios.io - All-in-one Google Analytics, Pixels and Product Feed Manager for WooCommerce: by Conversios – 7.1.3
Faire for WooCommerce: by Faire – 1.7.3
Relevanssi: by Mikko Saari – 4.22.2
SVG Support: by Benbodhi – 2.5.5
WooCommerce PayPal Payments: by WooCommerce – 2.8.2

### Dropin Plugins () ###

maintenance.php: maintenance.php
object-cache.php: object-cache.php

### Must Use Plugins (1) ###

Code Profiler (mu-plugin): by Jerome Bruandet – 1.0

### Settings ###

API Enabled: –
Force SSL: –
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 160 KB

### WC Pages ###

Shop base: ❌ Page ID is set
but the page does not exist

Cart: ❌ Page does not contain the [woocommerce_cart] shortcode or the woocommerce/cart block.
Checkout: ❌ Page does not contain the [woocommerce_checkout] shortcode or the woocommerce/checkout block.
My account: #1070 - /my-account-2/
Terms and conditions: ❌ Page not set

### Theme ###

Name: Avada Child
Version: 1.0.1
Author URL: https://theme-fusion.com
Child Theme: ✔
Parent Theme Name: Avada
Parent Theme Version: 7.11.9
Parent Theme Author URL: https://themeforest.net/user/ThemeFusion
WooCommerce Support: ✔

### Templates ###

Overrides: Avada-Child-Theme/woocommerce/cart/cart-shipping.php version 3.6.0 is out of date. The core version is 8.8.0
Avada/woocommerce/cart/cart.php
Avada/woocommerce/checkout/form-pay.php
Avada-Child-Theme/woocommerce/checkout/payment-method.php
Avada-Child-Theme/woocommerce/checkout/payment.php version 3.5.3 is out of date. The core version is 8.1.0
Avada/woocommerce/checkout/review-order.php
Avada-Child-Theme/woocommerce/emails/customer-completed-order.php
Avada/woocommerce/single-product/short-description.php
Avada/woocommerce/single-product/tabs/additional-information.php
Avada/woocommerce/single-product/tabs/description.php

Outdated Templates: ❌
					
					
						Learn how to update

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: experimental-blocks
minified-js
pattern-toolkit-full-composability
product-pre-publish-modal
printful
settings
async-product-editor-category-field
product-editor-template-system

Daily Cron: ✔ Next scheduled: 2024-07-30 09:35:00 +10:00
Options: ✔
Notes: 185
Onboarding: completed

### Action Scheduler ###

Canceled: 31
Oldest: 2024-07-24 20:17:32 +0000
Newest: 2024-07-24 17:31:10 +0000

Complete: 612
Oldest: 2024-07-08 17:42:24 +0000
Newest: 2024-07-29 12:29:22 +0000

Failed: 2,993
Oldest: 2022-09-14 14:38:13 +0000
Newest: 2024-07-26 20:19:37 +0000

Pending: 3
Oldest: 2024-07-29 20:03:30 +0000
Newest: 2024-07-29 20:45:24 +0000


### Status report information ###

Generated at: 2024-07-29 22:35:10 +10:00
```
</details>

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by gigitux** [comment]
Hi @rudolfl! Thanks for opening the issue!

I tried to replicate this issue with:
- storefront
- sample products

And I wasn't able to replicate it:


https://github.com/user-attachments/assets/ebc6da95-c821-44c8-9d74-db88a6fa3c32

Can you try to disable all the plugins enabled and try again? Thanks!

**Comment by rudolfl** [comment]
Hi Luigi

Thank you for looking into this.
I did the same test again -- disabled all plugins and switched to
StoreFront. Cant see the problem. Further investigations located a
problematic plugin (it is only relevant in Avada theme context, so this is
quite strange in itself, as it should be operational).

I guess, when I tried last time, some JS was still loading from cache
causing a problem. This time, I did the test and then loaded the front end
in the incognito mode.

Thank you again for your time

Rudolf
